modules/pipeline/pipengine/actionexecutor/plugins/wait/wait.go

```diff
@@ -0,0 +1,191 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wait
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/modules/pipeline/pipengine/actionexecutor/types"
+	"github.com/erda-project/erda/modules/pipeline/spec"
+	"github.com/erda-project/erda/pkg/envconf"
+)
+
+var Kind = types.Kind(spec.PipelineTaskExecutorKindWait)
+
+func init() {
+	types.MustRegister(Kind, func(name types.Name, options map[string]string) (types.ActionExecutor, error) {
+		return &Wait{
+			name:    name,
+			options: options,
+		}, nil
+	})
+}
+
+type Wait struct {
+	name    types.Name
+	options map[string]string
+}
+
+func (w *Wait) Kind() types.Kind {
+	return Kind
+}
+
+func (w *Wait) Name() types.Name {
+	return w.name
+}
+
+func (w *Wait) Exist(ctx context.Context, task *spec.PipelineTask) (bool, bool, error) {
+	status := task.Status
+	switch true {
+	case status == apistructs.PipelineStatusAnalyzed, status == apistructs.PipelineStatusBorn:
+		return false, false, nil
+	case status == apistructs.PipelineStatusCreated:
+		return true, false, nil
+	case status == apistructs.PipelineStatusQueue, status == apistructs.PipelineStatusRunning:
+		return true, true, nil
+	case status.IsEndStatus():
+		return true, true, nil
+	default:
+		return false, false, fmt.Errorf("invalid status when query task exist")
+	}
+}
+
+func (w *Wait) Create(ctx context.Context, task *spec.PipelineTask) (interface{}, error) {
+	return nil, nil
+}
+
+func (w *Wait) Start(ctx context.Context, task *spec.PipelineTask) (interface{}, error) {
+	created, started, err := w.Exist(ctx, task)
+	if err != nil {
+		return apistructs.PipelineStatusDesc{}, err
+	}
+
+	if !created {
+		logrus.Warnf("wait: action not create yet, try to create, pipelineID: %d, taskID: %d", task.PipelineID, task.ID)
+		_, err = w.Create(ctx, task)
+		if err != nil {
+			return nil, err
+		}
+		logrus.Warnf("scheduler: action created, continue to start, pipelineID: %d, taskID: %d", task.PipelineID, task.ID)
+	}
+
+	if started {
+		logrus.Warnf("scheduler: action already started, pipelineID: %d, taskID: %d", task.PipelineID, task.ID)
+		return nil, nil
+	}
+
+	executorCh := ctx.Value(spec.CtxExecutorChKey).(chan interface{})
+	if executorCh == nil {
+		return nil, errors.Errorf("wait: failed to get exector channel, pipelineID: %d, taskID: %d", task.PipelineID, task.ID)
+	}
+
+	waitSec, err := w.getWaitSec(task)
+	if err != nil {
+		return nil, err
+	}
+
+	timer := time.NewTimer(time.Duration(waitSec) * time.Second)
```

Comment:
Bad.
How about already passed the waitTime if pipeline broken for a long time?

Reply:
when execute status interface, will get success status if passed the waitTIme